services/app-api/resolvers/fetchSubmission2.test.ts

```diff
@@ -0,0 +1,299 @@
+        const server = await constructTestPostgresServer()
+
+        // First, create a new submission
+        const stateSubmission = await createTestStateSubmission(server)
+
+        const createdID = stateSubmission.id
+
+        // then see if we can fetch that same submission
+        const input = {
+            submissionID: createdID,
+        }
+
+        const result = await server.executeOperation({
+            query: FETCH_SUBMISSION_2,
+            variables: { input },
+        })
+
+        expect(result.errors).toBeUndefined()
+
+        const resultSub = result.data?.fetchSubmission2.submission
+        expect(resultSub.id).toEqual(createdID)
+        expect(resultSub.revisions.length).toEqual(1)
+
+        const revision = resultSub.revisions[0].revision
+
+        const subData = base64ToDomain(revision.submissionData)
+        if (subData instanceof Error) {
+            throw subData
+        } 
+
+        expect(subData.id).toEqual(createdID)
+        expect(subData.programIDs).toEqual(['cnet'])
+        expect(subData.submissionDescription).toEqual('An updated submission')
+        expect(subData.documents).toEqual([])
+        expect(subData.contractDocuments).toEqual([
+            {
+                name: 'contractDocument.pdf',
+                s3URL: 'fakeS3URL',
+                documentCategories: ['CONTRACT'],
+            },
+        ])
+    })
+
+    it('returns nothing if the ID doesnt exist', async () => {
+        const server = await constructTestPostgresServer()
+
+        // then see if we can fetch that same submission
+        const input = {
+            submissionID: 'BOGUS-ID',
+        }
+
+        const result = await server.executeOperation({
+            query: FETCH_SUBMISSION_2,
+            variables: { input },
+        })
+
+        expect(result.errors).toBeUndefined()
+
+        const resultSub = result.data?.fetchSubmission2.submission
+        expect(resultSub).toBeNull()
+    })
+
+    it('returns multiple submissions payload with multiple revisions', async () => {
+        const server = await constructTestPostgresServer()
+
+        const cmsServer = await constructTestPostgresServer({
+            context: {
+                user: {
+                    name: 'Zuko',
+                    role: 'CMS_USER',
+                    email: 'zuko@example.com',
+                },
+            },
+        })
+
+        // First, create a new submission
+        const stateSubmission = await createTestStateSubmission(server)
+        const createdID = stateSubmission.id
+
+        // unlock it
+        await unlockTestDraftSubmission(cmsServer, createdID)
+
+        // then see if we can fetch that same submission
+        const input = {
+            submissionID: createdID,
+        }
+
+        const result = await server.executeOperation({
+            query: FETCH_SUBMISSION_2,
+            variables: { input },
+        })
+
```

Comment:
How about adding a third revision here as well and then we can also quick check that the orderBy is right across each position as well.